analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -206,6 +206,45 @@
   r.nid IS NULL
 """
 
+
+sql40 = """
+SELECT
+    wid,
+    nid,
+    ST_AsText(geom),
+    highway
+FROM (
+SELECT
+    MIN(way_ends.id) AS wid,
+    nodes.id AS nid,
+    nodes.geom,
+    MIN(way_ends.highway) AS highway
+FROM
+    {0}highway_ends AS way_ends
+    JOIN highways ON
+        highways.linestring && way_ends.linestring AND
+        way_ends.nid = ANY(highways.nodes) AND
+        NOT highways.is_construction AND
+        NOT highways.is_oneway AND -- oneway implies an exit
+        highways.tags->'service' = 'drive-through'
```

Comment:
You need the full network to check connectiveness.

Reply:
You nee to filter the highway_ends, not the highway.

Reply:
Thank you! I see what you mean. 
I tried to implement it differently now, but I don't know if the new method is correct